Fix: Add error handling with custom exceptions

mtache opened the pull request:
## Change Summary

Added custom exception and error handling to get_container_id method

## Related Issue(s)

Fixes #433

## Component(s) name

`arista.cvp.container_tools_v3`
`arista.cvp.exceptions`
`pytest`

## Proposed changes
When interacting with CVP via cvprac, 2 situations can happen:
1. Ressource does not exist on CVP instance, cvprac methods return None in most (all) cases. If not handled, this can result in `TypeError` with no information for the end-user. For this scenario a `AnsibleCVPNotFoundError` exception has been created.
2. CVP API has changed and cvprac returns a different data structure the code does not correctly handle. This could result in different errors depending on the implementation `TypeError` if the code returns None, `KeyError` if the code tries to access a dict element without checking if it exists... Similarly, this gives no information to the end-user and increase the time needed for maintainers to debug the code and adapt to the new changes. For this scenario, a `AnsibleCVPApiError` exception has been created.

As stated by [Ansible guidelines](https://ansible-docs.readthedocs.io/zh/stable-2.0/rst/developing_modules.html#module-checklist), exceptions are bugs and needs to be handled internally (i.e. caught and fails with fail_json method).

The benefits of using exceptions here will be to ease debug on API change and gives useful messages to the end-user through fail_json.

This has currently only be implemented for get_container_id.

## How to test
`make test TESTS=unit/test_cv_container_tools.py`

```
=========================== short test summary info ============================
PASSED unit/test_cv_container_tools.py::TestCvContainerTools::test_get_container_id[Tenant-root]
PASSED unit/test_cv_container_tools.py::TestCvContainerTools::test_get_container_id[ansible-cvp-tests-1-container_274787b2-c45a-4b0c-8b97-eb4011909b02]
PASSED unit/test_cv_container_tools.py::TestCvContainerTools::test_get_container_id[Simple Fabric-container_6ac90bc0-f6d4-436f-bff0-8c064c247f01]
PASSED unit/test_cv_container_tools.py::TestCvContainerTools::test_get_container_id_not_found
PASSED unit/test_cv_container_tools.py::TestCvContainerTools::test_get_container_id_bad_key
============================== 5 passed in 1.45s ===============================
```

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- Review with core team
- Eventually implement for other methods

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

mtache committed:
Fix(cv_container_v3): CvContainerTools could be instantiated without an AnsibleModule instance

This could lead to TypeError exception when the code reaches self.__ansible.fail_json calls making failing test case coding impossible.

mtache committed:
Pytest: Add ARISTA_AVD_CVAAS environment variable

mtache committed:
Pytest: Pass the current log level to cvprac.cvp_client.CvpClient class

mtache committed:
Pytest: Add mock classes for AnsibleModule and CvpClient

mtache committed:
Pytest(cv_container_v3): Add unit test cases for get_container_id

titom73 commented:
Looks a great first PR! Thanks for your contribution!

This PR will be reviewed after PR #426 as both update Pytest mechanisms

mtache committed:
Fix: Add custom exceptions classes

mtache committed:
Fix(cv_container_v3): Add error handling to get_container_id

- AnsibleCVPNotFoundError
	Raised when container is not found in CVP instance
- AnsibleCVPApiError
	Raised when expected API data structure could not be read

mtache committed:
Linting(cv_container_v3): Configure pycodestyle correctly and fix ansible-test sanity errors

mtache committed:
Test(cv_container_v3): Update get_container_id unit test to use mock classes

titom73 reviewed (approved):
LGTM!

titom73 committed:
Merge branch 'devel' into issues/fix-cv-container-v3
